src/YieldDisburser.sol

```diff
@@ -282,64 +267,36 @@ contract YieldDisburser is OwnableUpgradeable {
      * @param _percentages List of percentages as integers for each project
      */
     function castVote(uint256[] calldata _percentages) public {
-        if (
-            this.getVotingPowerForPeriod(block.number - (minHoldingDuration / blockTime), block.number, msg.sender)
-                < minRequiredVotingPower
-        ) revert BelowMinRequiredVotingPower();
-
-        _castVote(msg.sender, _percentages);
+        if (holderToLastVoted[msg.sender] > lastClaimedTimestamp) revert AlreadyVotedInCycle();
+        uint256 votingPower =
+            this.getVotingPowerForPeriod(lastClaimedBlockNumber - cycleLength, lastClaimedBlockNumber, msg.sender);
+        if (votingPower < minRequiredVotingPower) revert BelowMinRequiredVotingPower(minRequiredVotingPower);
+        _castVote(msg.sender, _percentages, votingPower);
     }
 
     /**
      * @notice Internal function for casting votes for a specified user
      * @param _account Address of user to cast votes for
      * @param _points Basis points for calculating the amount of votes cast
      */
-    function _castVote(address _account, uint256[] calldata _points) internal {
+    function _castVote(address _account, uint256[] calldata _points, uint256 _votingPower) internal {
         uint256 length = projects.length;
         if (_points.length != length) revert IncorrectNumberOfProjects();
 
-        if (holderToDistribution[_account].length > 0) {
-            delete holderToDistribution[_account];
-        } else {
-            voters.push(_account);
-        }
-        currentVotes++;
-        holderToDistribution[_account] = _points;
         uint256 total;
         for (uint256 i; i < length; ++i) {
             if (_points[i] > maxPoints) revert VotePointsTooLarge();
```

Comment:
Is there no "maximum sum of points" that a user can define? Only a maximum individual point value regardless of how many projects there are?? I'm not following this logic. Are we not asking users to divide their vote into a distribution out of 100% with a specific level of precision? If so, we should ask them for an array of values where the length = the number of projects and all the values add up to 10000 (assuming two decimals of precision).

Reply:
Basically what we and @subject026  agreed on was that they would be able to send points any way they want up to maxPoints, so all the values can only add up to projectsLength * maxPoints indeed. We do check that the array is projectsLength long. I think it is the same as asking users to divide their vote into a distribution out specific level of precision here is maxPoints essentially. that's the precision for 1 distribution

Reply:
Originally users were to be able to change their vote so we wanted to be able to retrieve the original point values they input when voting.